src/mame/includes/pgm.h

```diff
@@ -69,61 +70,88 @@ class pgm_state : public driver_device
-	DECLARE_WRITE16_MEMBER(m68k_l1_w);
+	DECLARE_WRITE8_MEMBER(m68k_l1_w);
 	DECLARE_WRITE8_MEMBER(z80_l3_w);
-	DECLARE_WRITE16_MEMBER(pgm_tx_videoram_w);
-	DECLARE_WRITE16_MEMBER(pgm_bg_videoram_w);
-
-	void init_pgm();
-
-	TILE_GET_INFO_MEMBER(get_pgm_tx_tilemap_tile_info);
-	TILE_GET_INFO_MEMBER(get_pgm_bg_tilemap_tile_info);
-	DECLARE_VIDEO_START(pgm);
-	DECLARE_MACHINE_START(pgm);
-	DECLARE_MACHINE_RESET(pgm);
-	uint32_t screen_update_pgm(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect);
-	DECLARE_WRITE_LINE_MEMBER(screen_vblank_pgm);
-	TIMER_DEVICE_CALLBACK_MEMBER(pgm_interrupt);
-
-	inline void pgm_draw_pix( int xdrawpos, int pri, uint16_t* dest, uint8_t* destpri, uint16_t srcdat);
-	inline void pgm_draw_pix_nopri( int xdrawpos, uint16_t* dest, uint8_t* destpri, uint16_t srcdat);
-	inline void pgm_draw_pix_pri( int xdrawpos, uint16_t* dest, uint8_t* destpri, uint16_t srcdat);
-	void draw_sprite_line( int wide, uint16_t* dest, uint8_t* destpri, int xzoom, int xgrow, int flip, int xpos, int pri, int realxsize, int palt, int draw );
-	void draw_sprite_new_zoomed( int wide, int high, int xpos, int ypos, int palt, int flip, bitmap_ind16 &bitmap, bitmap_ind8 &priority_bitmap, uint32_t xzoom, int xgrow, uint32_t yzoom, int ygrow, int pri );
-	void draw_sprite_line_basic( int wide, uint16_t* dest, uint8_t* destpri, int flip, int xpos, int pri, int realxsize, int palt, int draw );
-	void draw_sprite_new_basic( int wide, int high, int xpos, int ypos, int palt, int flip, bitmap_ind16 &bitmap, bitmap_ind8 &priority_bitmap, int pri );
-	void draw_sprites( bitmap_ind16& spritebitmap, uint16_t *sprite_source, bitmap_ind8& priority_bitmap );
+	void pgm_tx_videoram_w(offs_t offset, u16 data, u16 mem_mask);
+	void pgm_bg_videoram_w(offs_t offset, u16 data, u16 mem_mask);
+
+	TILE_GET_INFO_MEMBER(get_tx_tilemap_tile_info);
+	TILE_GET_INFO_MEMBER(get_bg_tilemap_tile_info);
+	u32 screen_update(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect);
+	DECLARE_WRITE_LINE_MEMBER(screen_vblank);
+	TIMER_DEVICE_CALLBACK_MEMBER(interrupt);
+
+	inline void pgm_draw_pix(u16 const xdrawpos, u8 const pri, u16* dest, u8* destpri, const rectangle &cliprect, u16 const srcdat);
+	inline void pgm_draw_pix_nopri(u16 const xdrawpos, u16* dest, u8* destpri, const rectangle &cliprect, u16 const srcdat);
+	inline void pgm_draw_pix_pri(u16 const xdrawpos, u16* dest, u8* destpri, const rectangle &cliprect, u16 const srcdat);
+
+	void draw_sprite_line(
+	u16 const wide,
+	u16* dest, u8* destpri, const rectangle &cliprect,
+	u32 const xzoom, bool const xgrow,
+	u8 const flip, s16 const xpos,
+	u8 const pri,
+	u16 const realxsize,
+	u8 const palt,
+	bool const draw);
+
+	void draw_sprite_new_zoomed(
+	u16 const wide, u16 const high,
+	s16 const xpos, s16 const ypos,
+	u8 const palt, u8 const flip,
+	bitmap_ind16 &bitmap, bitmap_ind8 &priority_bitmap, const rectangle &cliprect,
+	u32 const xzoom, bool const xgrow, u32 const yzoom, bool const ygrow,
```

Comment:
You should indent line continuations by two levels to make it clear where they start/end, for example:

```
func(
        T1 p1,
        T2 p2);
```

The trouble with making formal value parameters `const` is it doesn't affect the function signature but it still requires the header to be changed if you decide you want to modify the value in the function for some reason.  (Changes to value parameters are local to the function and don't affect the caller.)  It's not really going to be a problem in a driver header like this, because it's only going to cause the PGM driver files to be recompiled anyway, but it's definitely worth avoiding for widespread devices.  (Reference/pointer parameters should always be `const` if the function isn't using them as outputs, as the caller can be affected in that case.)